src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Definitions/GoToDefinitionHandler.cs

```diff
@@ -159,10 +159,12 @@ private static async Task<LSP.Location[]> GetSymbolDefinitionLocationsAsync(Xaml
             var items = NavigableItemFactory.GetItemsFromPreferredSourceLocations(context.Solution, symbol, displayTaggedParts: null, cancellationToken);
             if (items.Any())
             {
+                RoslynDebug.AssertNotNull(context.Solution);
                 foreach (var item in items)
                 {
+                    var document = await item.Document.GetDocumentAsync(context.Solution, cancellationToken).ConfigureAwait(false);
                     var location = await ProtocolConversions.TextSpanToLocationAsync(
-                        item.Document, item.SourceSpan, item.IsStale, cancellationToken).ConfigureAwait(false);
+                        document, item.SourceSpan, item.IsStale, cancellationToken).ConfigureAwait(false);
```

Comment:
i don't think this is safe anymore.  item.IsStale was previously used to say that the result info might not match teh current snapshot.  But now that you're not rooting an actual snapshot, it may always be the case that things may be stale (e.g. spans map to things that are now no longer legal).

Reply:
➡️ This thread is covered by https://github.com/dotnet/roslyn/pull/68068#discussion_r1185325428